### PR TITLE
Feature n7s tmcc#521 support for append

### PR DESCRIPTION
**Pull request by rphh**

### CODE REVIEW

File: tests/unit/Makefile [hunk 1]
```diff
@@ -4,7 +4,8 @@ TESTS = \
 	sdl/test_model.py \
 	promela/test_model.py \
 	translator/test_translator.py \
-	stop_condition/test_scl_parser.py
+	stop_condition/test_scl_parser.py \
+    translator_append_operator/test_translator_append_operator.py
```
Comment (Lurkerpas) [inline tests/unit/Makefile:8]: Indent

File: sdl2promela/translator.py [hunk 2]
```diff
@@ -951,6 +928,8 @@ def __generate_expression(context: Context, expression: sdlmodel.BinaryExpressio
             )
             .build()
         )
+    elif expression.operator == sdlmodel.BinaryOperator.APPEND:
+        raise NotImplementedError("binary APPEND operator not supported here")
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:932]: Please provide a more descriptive error message.

File: sdl2promela/translator.py [hunk 3]
```diff
@@ -1729,6 +1710,212 @@ def __generate_statement(
...
+def __generate_assignment_with_append(
+    context: Context,
+    left: Union[
+        sdlmodel.VariableReference, sdlmodel.ArrayAccess, sdlmodel.MemberAccess
+    ],
+    right: sdlmodel.Expression,
+    left_type: type,
+):
+    destination = __generate_variable_name(context, left, True)
+    destination_type = resolve_asn1_type(context.sdl_model.types, left.type)
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:1806]: This method is too long. Please split into descriptively named submethods.

File: sdl2promela/translator.py (hunk 2 above)
Comment (rphh) [reply sdl2promela/translator.py]: changed

File: sdl2promela/translator.py (hunk 3 above)
Comment (rphh) [reply sdl2promela/translator.py:1861]: I've done some refactor. One function was extracted and I created a helper function, so creation of assignment statement is shorter and more readable IMO.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**